pkg/operator/encryption/key_controller_encryption_test.go

```diff
@@ -0,0 +1,154 @@
+package encryption
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/client-go/kubernetes/fake"
+	clientgotesting "k8s.io/client-go/testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	encryptionSecretKeyDataForTest = "encryption.operator.openshift.io-key"
+)
+
+func TestEncryptionKeyController(t *testing.T) {
+	scenarios := []struct {
+		name                     string
+		initialSecrets           []runtime.Object
+		encryptionSecretSelector metav1.ListOptions
+		targetNamespace          string
+		targetGRs                map[schema.GroupResource]bool
+		validateFunc             func(actions []clientgotesting.Action, targetNamespace string, targetGRs map[schema.GroupResource]bool)
+	}{
+		// scenario 1: assumes a clean slate, that is, there are no previous resources in the system.
+		// It expects that a secret resource with an appropriate key, name and labels will be created.
+		{
+			name: "scenario 1: checks if a secret with AES256 key for core/secret is created",
```

Comment:
nit: I use not to number scenarios. There will always be another one to insert between existing ones. But just my taste :)